### PR TITLE
MM-28922: what are incidents landing page

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
When wide enough:
image

and when not so wide (and themed differently):
image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28922

### CODE REVIEW

File: webapp/src/components/backstage/incidents/incident_list/incident_list.tsx [hunk 1]
```diff
@@ -49,11 +51,102 @@ const ControlComponent = (ownProps: ControlProps<any>) => (
     </div>
 );
 
+const NoContentContainer = styled.div`
```
Comment (lieut-data) [inline webapp/src/components/backstage/incidents/incident_list/incident_list.tsx:54]: /Some/ of this was lifted from the playbook version of same, then tweaked to make it work with different browser widths. I'll probably lift this back to playbooks later on, but for now just copying in place vs. trying to refactor out.

File: webapp/src/components/backstage/incidents/incident_list/incident_list.tsx [hunk 2]
```diff
@@ -49,11 +51,102 @@ const ControlComponent = (ownProps: ControlProps<any>) => (
...
+                <NoContentIncidentSvg/>
+            </NoContentIncidentSvgContainer>
+        </NoContentContainer>
+    );
+};
+
 const BackstageIncidentList: FC = () => {
-    const [incidents, setIncidents] = useState<Incident[]>([]);
+    const dispatch = useDispatch();
+    const [showNoIncidents, setShowNoIncidents] = useState(false);
```
Comment (lieut-data) [inline webapp/src/components/backstage/incidents/incident_list/incident_list.tsx:146]: Explicitly tracking `showNoIncidents` so I can make a decision on first mount vs. dealing with the transition off of filtering to an empty list that "flashes" momentarily.

File: webapp/src/components/backstage/incidents/incident_list/incident_list.tsx [hunk 3]
```diff
@@ -215,57 +335,45 @@ const BackstageIncidentList: FC = () => {
...
-                    !incidents.length && !isFiltering &&
-                    <div className='text-center pt-8'>
-                        {'There are no incidents for '}
-                        <i>{currentTeam.display_name}</i>
-                        {'.'}
-                    </div>
-                }
-                {
-                    !incidents.length && isFiltering &&
+                {incidents.length === 0 &&
```
Comment (lieut-data) [inline webapp/src/components/backstage/incidents/incident_list/incident_list.tsx:338]: Replaced by this PR's content.

File: webapp/src/components/backstage/incidents/incident_list/incident_list.tsx [hunk 4]
```diff
@@ -215,57 +335,45 @@ const BackstageIncidentList: FC = () => {
...
                     <div className='text-center pt-8'>
                         {'There are no incidents for '}
                         <i>{currentTeam.display_name}</i>
                         {' matching those filters.'}
                     </div>
                 }
-
-                {
-                    incidents.map((incident) => (
+                {incidents.map((incident) => (
```
Comment (lieut-data) [inline webapp/src/components/backstage/incidents/incident_list/incident_list.tsx:345]: Mostly indentation differences here. I also ripped out the unnecessary `Route` stuff.

File: webapp/src/components/backstage/incidents/incident_list/incident_list.tsx [hunk 5]
```diff
@@ -49,11 +51,102 @@ const ControlComponent = (ownProps: ControlProps<any>) => (
...
+        display: none;
+    }
+`;
+
+const NoContentPage = (props: {onNewIncident: () => void}) => {
+    return (
+        <NoContentContainer>
+            <NoContentTextContainer>
+                <NoContentTitle>{'What are Incidents?'}</NoContentTitle>
+                <NoContentDescription>{'Incidents are unexpected situations which impact business operations which require an immediate, multi-disciplinary, response and benefit from a clearly defined process. When the situation is resolved, the incident is ended, and the playbook can be updated to improve the response to similar incidents in the future.'}</NoContentDescription>
```
Comment (agarciamontoro) [inline webapp/src/components/backstage/incidents/incident_list/incident_list.tsx:128]: Nit: I feel like those two "which" here are really close to each other.

File: webapp/src/components/backstage/incidents/incident_list/incident_list.tsx [hunk 6]
```diff
@@ -49,11 +51,102 @@ const ControlComponent = (ownProps: ControlProps<any>) => (
...
+                <NoContentButton
+                    className='mt-6'
+                    onClick={() => props.onNewIncident()}
```
Comment (agarciamontoro) [inline webapp/src/components/backstage/incidents/incident_list/incident_list.tsx:131]: Ultra nit:
```suggestion
                    onClick={props.onNewIncident}
```

**Review by agarciamontoro — APPROVED** [review]
A couple of nits below, but everything looks fine! And the illustrations are so cool, @asaadmahmood! :tada:

**Comment by lieut-data** [comment]
Awesome, thanks @agarciamontoro! Feedback applied.

**Pull request merged**